src/mio_wrapper.rs

```diff
@@ -0,0 +1,111 @@
+use std::io;
+use std::os::unix::io::{RawFd, FromRawFd};
+
+use libc::{self, c_void};
+use mio::{Poll, Token, Ready, PollOpt, Evented};
+use mio::unix::EventedFd;
+
+pub struct Tun(RawFd);
+pub struct Tap(RawFd);
```

Comment:
Well, TcpStream has the same implementation too, but for some reason they decided not to provide it generically.

This is actually why type system exists. Using File here is a hack. It's not a file. We shouldn't use file for anything on top of RawFd. For example there is no sense in executing `seek` for this fd. And we will probably add some ioctl wrappers on this object.

At the end of the day, there is no point in optimizing one line of code which does `libc::close`.